### PR TITLE
More info when critical failure occurs

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

Log problematic module to help with debugging critical failure.
Right now it's a real guess-work.
Related to #11088.

### CODE REVIEW

**Review by Akirathan — APPROVED** [review]

File: lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java [hunk 1]
```diff
@@ -131,7 +132,12 @@ public Tree parse(CharSequence input) {
     var metadata = getMetadata(state);
     serializedTree.order(ByteOrder.LITTLE_ENDIAN);
     var message = new Message(serializedTree, input, base, metadata);
-    return Tree.deserialize(message);
+    try {
+      return Tree.deserialize(message);
+    } catch (BufferUnderflowException | IllegalArgumentException e) {
+      System.err.println("Unrecoverable parser failure for: " + input);
```
Comment (JaroslavTulach) [inline lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java]: Can we use our logging framework, please?

**Review by JaroslavTulach — DISMISSED** [review]

File: lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java (hunk 1 above)
Comment (hubertp) [reply lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java]: Fair enough

**Comment by JaroslavTulach** [comment]
There are still some [CI failures](https://github.com/enso-org/enso/actions/runs/10940656341/job/30373557349?pr=11092#step:7:5510).

**Review by GregoryTravis — APPROVED** [review]

File: build/build/src/rust/parser.rs [hunk 2]
```diff
@@ -63,7 +79,11 @@ pub async fn run_self_tests(repo_root: &RepoRoot) -> Result {
 
     Javac
         .cmd()?
-        .apply(&javac::Classpath::new([lib.as_path(), base.as_path()]))
+        .apply(&javac::Classpath::new([
+            lib.as_path(),
+            base.as_path(),
+            Path::new(&dependencies_from_string),
```
Comment (kazcw) [inline build/build/src/rust/parser.rs]:
```suggestion
            Path::new(dependencies_from_string),
```

File: build/build/src/rust/parser.rs [hunk 3]
```diff
@@ -50,6 +51,21 @@ pub async fn generate_java(repo_root: &RepoRoot) -> Result {
...
+        .join("streams")
+        .join("runtime")
+        .join("managedClasspath")
+        .join("_global")
+        .join("streams")
+        .join("export");
+
+    let path = external_dependencies_file.into_os_string();
+    let mut dependencies_from_string = fs::read_to_string(path.clone())?;
+    dependencies_from_string.pop(); // Remove newline sign
```
Comment (kazcw) [inline build/build/src/rust/parser.rs]:
```suggestion
    let dependencies_from_string = fs::read_to_string(&external_dependencies_file)?;
    let dependencies_from_string = dependencies_from_string.trim_ascii_end();
```

**Review by kazcw — APPROVED** [review]

File: lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java [hunk 4]
```diff
@@ -131,7 +133,13 @@ public Tree parse(CharSequence input) {
     var metadata = getMetadata(state);
     serializedTree.order(ByteOrder.LITTLE_ENDIAN);
     var message = new Message(serializedTree, input, base, metadata);
-    return Tree.deserialize(message);
+    try {
+      return Tree.deserialize(message);
+    } catch (BufferUnderflowException | IllegalArgumentException e) {
+      LoggerFactory.getLogger(this.getClass())
+          .error("Unrecoverable parser failure for: {}", input, e);
```
Comment (JaroslavTulach) [inline lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java:140]: This code is calling [error(Object,Throwable)](https://www.slf4j.org/apidocs-2.1.0/org/slf4j/Logger.html#error(java.lang.String,java.lang.Throwable)) method. What's its behavior in Enso?

#### When running `runtime-integration-tests`

Will it print the message as well as exception to the (CI) console?

#### When running in `enso` CLI mode

Will it print the message as well as exception to the (CI) console? Will it do something different?

#### When running in Enso IDE mode

What will appear in console and what will appear in associated log file?

**Review by JaroslavTulach — COMMENTED** [review]
I'd like to understand what will be the behavior of the newly added `error(...)` call in various environments Enso code runs in. Thank you.

**Review by jdunkerley — APPROVED** [review]

File: lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java (hunk 4 above)
Comment (hubertp) [reply lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java:140]: This PR changes nothing when it comes to how log messages are consumed, this is calling a plain slf4j api. The actual printing is dependent on the implementation/configuration, as you know.

If you don't like the format/output of the logger in CLI or tests then please create a ticket for that. IDE works as expected.

Comment (JaroslavTulach) [reply lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java:140]: > If you don't like the format/output of the logger in CLI or tests then please create a ticket for that. IDE works as expected.

Which is?

> What will appear in console and what will appear in associated log file?

Will you, please, answer my question?

**Pull request merged**